### ISSUE TITLE
Support creating an installer from `InstallOS.pkg`

### ISSUE THREAD

**Issue by madsmtm** [BITespresso/createinstalliso#12]
The macOS 10.12 link (and probably the others) on [Apple's official download page](https://support.apple.com/en-us/102662) ends up downloading `InstallOS.dmg`, which in turn contains `InstallOS.pkg`, which is fairly troublesome to convert to the `Install macOS Sierra.app` that `createinstalliso` expects.

What's the recommended approach for these?

From my own testing, I ended up being able to create the iso after following the steps below (though not sure it actually runs yet):
1. Open `InstallOS.dmg` to get access to `InstallOS.pkg`
2. Unpack `InstallOS.pkg` and then `InstallOS.pkg/Payload` using instructions in https://stackoverflow.com/questions/11298855/how-to-unpack-and-pack-pkg-file (or maybe just `pkgutil --expand-full`?)
3. Copy `InstallESD.dmg` to `Install XXX.app/Contents/SharedSupport`
4. Run `xattr -cr Install XXX.app` to allow codesign to work
5. Run `./createinstalliso`

**Comment by jgrisham** [comment]
Can you just skip steps 3-5 and just directly use the `InstallESD.dmg` image? *(Iirc, it is bootable EFI-compatible ISO and can be burned to a DVD or a USB drive.)*

Using Windows 11 and 7-Zip, for example, simply re-naming `InstallOS.pkg` to `InstallOS.xar` allowed opening it directly in 7-Zip, which then allowed extracting the `InstallESD.dmg` file. *(The `InstallOS.pkg` file itself was obtainable from the downloaded disk image file using [HFSExplorer](https://www.catacombae.org/hfsexplorer/) and [OpenJDK](https://learn.microsoft.com/en-us/java/openjdk/download).)*
 * On macOS, an application such as *[Suspicious Package](https://www.mothersruin.com/software/SuspiciousPackage/)* would also get this done.
 * Linux and macOS *Terminal* methods should also be farily straightforward.

---
Tl;dr:

*Renaming `InstallESD.dmg` to `InstallESD.iso` should allow it to be used directly with a much wider array of software.*

**Comment by madsmtm** [comment]
I had no idea what `InstallESD.dmg` was, but cool!

Are you sure that it's not necessary to invoke `createinstallmedia`?
